File: app/src/main/java/org/akvo/mockcaddisfly/data/result/DeviceFactory.java
package org.akvo.mockcaddisfly.data.result;

import android.os.Build;

import java.util.Locale;

public class DeviceFactory {

    public static Device createDefaultDevice() {
        return new Device(Build.PRODUCT, "Android - " + Build.VERSION.RELEASE + " ("
                + Build.VERSION.SDK_INT + ")", Build.MODEL, Locale.getDefault().getLanguage(),
                Build.MANUFACTURER, Locale.getDefault().getCountry());
    }
}

File: app/src/main/java/org/akvo/mockcaddisfly/data/result/DeviceFactory.kt
package org.akvo.mockcaddisfly.data.result

import android.os.Build

import java.util.Locale

object DeviceFactory {

    fun createDefaultDevice(): Device {
        return Device(Build.PRODUCT, "Android - " + Build.VERSION.RELEASE + " ("
                + Build.VERSION.SDK_INT + ")", Build.MODEL, Locale.getDefault().language,
                Build.MANUFACTURER, Locale.getDefault().country)
    }
}
